Mathlib/LinearAlgebra/Reflection.lean

```diff
@@ -114,6 +121,173 @@ lemma bijOn_reflection_of_mapsTo {Φ : Set M} (h : f x = 2) (h' : MapsTo (reflec
     BijOn (reflection h) Φ Φ :=
   (invOn_reflection_of_mapsTo h).bijOn h' h'
 
+/-! ### Powers of the product of two reflections
+
+Let $M$ be a module over a commutative ring $R$. Let $x, y \in M$ and $f, g \in M^*$ with
+$f(x) = g(y) = 2$. The corresponding reflections $r_1, r_2 \colon M \to M$ (`Module.reflection`) are
+given by $r_1z = z - f(z) x$ and $r_2 z = z - g(z) y$. These are linear automorphisms of $M$.
+
+To define reflection representations of a Coxeter group, it is important to be able to compute the
+order of the composition $r_1 r_2$.
+
+Note that if $M$ is a real inner product space and $r_1$ and $r_2$ are both orthogonal
+reflections (i.e. $f(z) = 2 \langle x, z \rangle / \langle x, x \rangle$ and
+$g(z) = 2 \langle y, z\rangle / \langle y, y\rangle$ for all $z \in M$),
+then $r_1 r_2$ is a rotation by the angle
+$$\cos^{-1}\left(\frac{f(y) g(x) - 2}{2}\right)$$
+and one may determine the order of $r_1 r_2$ accordingly.
+
+However, if $M$ does not have an inner product, and even if $R$ is not $\mathbb{R}$, then we may
+instead use the formulas in this section. These formulas all involve evaluating Chebyshev
+$S$-polynomials (`Polynomial.Chebyshev.S`) at $t = f(y) g(x) - 2$, and they hold over any
+commutative ring. -/
+section
+
+open Int Polynomial.Chebyshev
+
+variable {x y : M} {f g : Dual R M} (hf : f x = 2) (hg : g y = 2)
+
+local notation:max "t" => f y * g x - 2
+local notation:max "S_eval_t" n:max => Polynomial.eval t (S R n)
```

Comment:
I understand that these notations make it easier to work in this file. But outside this file they don't apply. Which means that many of the statements will become rather complicated, and not so easy to handle.

I am not completely sure what the best strategy is. But I think `S_eval_t` could maybe be a standalone definition. (But I'm not sure what the best name is for it.)

Reply:
What do you think of the following approach? Here I just add `t` as an extra assumption, together with `ht` giving its defining formula:
```
lemma reflection_mul_reflection_pow_apply (m : ℕ) (z : M) (t : R) (ht : t = f y * g x - 2) :
    ((reflection hf * reflection hg) ^ m) z =
      z +
        ((S R ((m - 2) / 2)).eval t * ((S R ((m - 1) / 2)).eval t + (S R ((m - 3) / 2)).eval t)) •
          ((g x * f z - g z) • y - f z • x) +
        ((S R ((m - 1) / 2)).eval t * ((S R (m / 2)).eval t + (S R ((m - 2) / 2)).eval t)) •
          ((f y * g z - f z) • x - g z • y) := by
```

Reply:
If we can convince the pretty printer to use dot-notation for `Polynomial.eval` (see [zulip](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Polynomial.2Eeval.20dot-notation.20delaborator)), then I think this is quite readable, and doesn't need local notation.